packages/code-du-travail-data/indexing/cdtnIds.js

```diff
@@ -0,0 +1,49 @@
+import * as XXH from "xxhashjs";
+
+// never change this seed, as ids are regenerated
+// each time but must remain stable
+const H = XXH.h64(0x1e7f);
+
+// default id length, can be longer (up to 16)
+const maxIdLength = 10;
+
+// use xxhash to hash source+id
+const hashId = (source, id, idLength) =>
+  // save 64bits hash as Hexa string up to maxIdLength chars (can be changed later in case of collision)
+  // as the xxhash function ensure distribution property
+  H.update(source + id)
+    .digest()
+    .toString(16)
+    .slice(0, idLength);
+
+const hashingSet = (hashMap, idLength) => (obj) => {
```

Comment:
```suggestion
const hashingSet = (...) => ({id, source}) => {
```
Plutôt que de prendre l'object en entier, et du coup tu retournerais le hash et ça serait à l'appelant de le mettre dans l'objet concerné. C'est une chouille plus flexible comme ça